pennylane_qiskit/converter.py

```diff
@@ -0,0 +1,194 @@
+    """Returns a PennyLane template created based on the input QuantumCircuit or QASM string.
+    Warnings are created for each of the QuantumCircuit instructions that were
+    not incorporated in the PennyLane template.
+
+    Args:
+        quantum_circuit (qiskit.QuantumCircuit): the QuantumCircuit to be converted
+
+    Returns:
+        function: the new PennyLane template
+    """
+
+    def _function(params: dict = None, wires: list = None):
+        """Returns a PennyLane template created based on the input QuantumCircuit.
+        Warnings are created for each of the QuantumCircuit instructions that were
+        not incorporated in the PennyLane template.
+
+        Args:
+            params (dict): specifies the parameters that need to bound in the QuantumCircuit
+
+            wires (Sequence[int] or int): The wires the converted template acts on.
+                Note that if the original QuantumCircuit acted on :math:`N` qubits,
+                then this must be a list of length :math:`N`.
+
+        Returns:
+            function: the new PennyLane template
+        """
+
+        qc = check_circuit_and_bind_parameters(quantum_circuit, params)
+
+        # Wires from a qiskit circuit are unique w.r.t. a register name and a qubit index
+        qc_wires = [(q.register.name, q.index) for q in quantum_circuit.qubits]
+
+        wire_map = map_wires(wires, qc_wires)
+
+        # Processing the dictionary of parameters passed
+        for op in qc.data:
+
+            instruction_name = op[0].__class__.__name__
+
+            operation_wires = [wire_map[(qubit.register.name, qubit.index)] for qubit in op[1]]
+
+            if instruction_name in inv_map and inv_map[instruction_name] in pennylane_ops.ops:
+
+                execute_supported_operation(inv_map[instruction_name], op[0].params, operation_wires)
+
+            else:
+                try:
+                    operation_matrix = op[0].to_matrix()
+                    pennylane_ops.QubitUnitary(operation_matrix, wires=operation_wires)
+                except (AttributeError, QiskitError):
+                    warnings.warn(__name__ + ": The {} instruction is not supported by PennyLane,"
+                                             " and has not been added to the template.".
+                                  format(instruction_name),
+                                  UserWarning)
+
+    return _function
+
+
+def load_qasm(qasm_string: str):
+    """Returns a PennyLane template created based on the input QASM string or file.
+
+        Args:
+            qasm_string (str): the name of the QASM file or string
+
+        Returns:
+            function: the new PennyLane template
+    """
+
+    # Checks whether a qasm string or a file name was provided
+    if isinstance(qasm_string, str) and not qasm_string.endswith('.qasm'):
+        return load(QuantumCircuit.from_qasm_str(qasm_string))
+
+    return load(QuantumCircuit.from_qasm_file(qasm_string))
```

Comment:
It might be best to separate this into two functions; `load_qasm_string` and `load_qasm_file`, just to be explicit. This avoids implicitly checking `qasm_string.endswith('.qasm')`, in case a user has saved the file with a different extension.

Reply:
That's true, there is actually no explicit requirement for the file to end with the string ".qasm". Thanks!